include/engine/datafacade/internal_memory_datafacade.hpp

```diff
@@ -0,0 +1,52 @@
+#ifndef INTERNAL_DATAFACADE_HPP
```

Comment:
IMHO: to avoid name clashes with any other possible libs, we should keep to `OSRM_ENGINE_DATAFACADE_INTERNAL_DATAFACADE_HPP` (full path header guards).